src/utils/format.ts

```diff
@@ -0,0 +1,16 @@
+export const camelToUnderscore = (key: string): string =>
+  key.replace(/([A-Z])/g, "_$1").toLowerCase();
+
+export function ObjectToJsonSnakeCaseKeys<OriginalType>(
+  original: OriginalType
+) {
+  const newObject = Object.fromEntries(
+    Object.keys(original as object)
+      .filter((key) => original[key as keyof OriginalType])
+      .map((key) => [
+        camelToUnderscore(key),
+        String(original[key as keyof OriginalType]),
+      ])
+  );
+  return newObject;
```

Comment:
I still beleive in explicit typings for the NASA service API rather than converting it using this logic.

Reply:
There are types and you can pass a type into this or you can use type assertion